### ISSUE TITLE
display:flex on fxFlex attribute ?

### ISSUE THREAD

**Issue by sambaptista** [zefoy/ngx-perfect-scrollbar#94]
Hello, 

I'm implementing your lib, but noticed something strange : when switching from a `<div fxFlex>` to a `<perfect-scrollbar fxFlex>` I got a new css instruction `display:flex` applied to my element.

Correct me if I'm wrong, but fxFlex is supposed to add `flex:1 1 auto` on child elements of a container with `display:flex`

The `display:flex` style is defined by `fxLayout` attribute that is destinated to containers and should not be applied by `fxFlex` attribute.

Actually, when styling a` <perfect-scrollbar>` as a child element, we apply container styles (and in my case, it breaks the layout).

<img width="366" alt="capture d ecran 2017-10-12 17 41 29" src="https://user-images.githubusercontent.com/5111137/31505299-e52f5c18-af74-11e7-85ce-4d364c2f424e.png">

Imho, the `display:flex` instruction should not be there, mostly if there is no flex-direction specified.

**Comment by sconix** [comment]
You are right it should not apply it if no fxLayout is not used. I took the easy way out adding the styles that work in our case since I haven't seen that many others using the flex-layout yet. I will fine tune the styles for next release during this week.

**Comment by sconix** [comment]
So do you think it will work for you if I separate the display / direction rules to be applied only if fxlayout attribute is used? Or actually I wonder if the correct fix would be just to apply all those when fxLayout is used. And not when fxFlex is applied since that itself should not do anything. So I think I just need to remove the fxFlex and fxFlexFill from the selectors and only apply those rules if fxLayout is used.

The use case where we need this is to be able to use flex when the content is not scrollable and it works nicely, I was just lazy to update the templates of our app so I added all those flex attributes.

**Comment by sambaptista** [comment]
I don't know if it will be enough, I'm investigating the topic and will bring you a more pragmatic answer (and jsfiddle) soon.

**Comment by sconix** [comment]
I don't see why it would not be enough if you are not using fxLayout on the ps component. Jsfiddle / Plunker would be nice to see your case so I can make sure that it works + our case works before releasing a fix. Will get back to this tomorrow.

**Comment by sconix** [comment]
I am thinking that this would be the "correct" style configuration:

```
  &[fxflex] {
    display: flex;
    flex-direction: column;
    -webkit-box-orient: column;
    -webkit-box-direction: column;
    height: auto;
    min-width: 0;
    min-height: 0;

    & > .ps {
      flex: 1 1 auto;
      -ms-flex: 1 1 auto;
      -webkit-box-flex: 1;
      width: auto;
      height: auto;
      min-width: 0;
      min-height: 0;
    }
  }

  &[fxlayout] {
    & > .ps,
    & > .ps > .ps-content {
      display: flex;
      flex: 1 1 auto;
      -ms-flex: 1 1 auto;
      -webkit-box-flex: 1;
      flex-direction: inherit;
      -webkit-box-orient: inherit;
      -webkit-box-direction: inherit;
      width: auto;
      height: auto;
      min-width: 0;
      min-height: 0;
    }
  }
```

So if the component has fxFlex then we use flex to style the ps container and if fxFlexLayout is used then we inherit the setup to the ps content so that contents can be layouted with flex. Initial testing shows this working, but there is one issue. For some reason the ps-content element gets row direction from the flex-layout, but I think this is a flex-layout bug. At least there has been issue about adding row incorrectly. Will study this bit more.

**Comment by sconix** [comment]
Ok, so one problem is that flex-layout / angular seems to set the fx* attributes only after it has set the styles. This causes the flex-layout set wrong styles due to the ps stylesheet styles not being in effect yet. So the fx* attributes can not be used to apply the styles, I need to add fx* inputs for the component and add additional classes based on those and then flex-layout will function properly and then its only about having the correct styles which I think should be like above.

EDIT: Seems that binding workaround wont work neither, I am starting to wonder if there is some flex-layout bug involved.

**Comment by sconix** [comment]
So finally I tracked the flex-layout problem to an open issue regarding ng-content which results in wrong layout styles to be set. This is in the beta.10 milestone so hopefully there will be fix for this soon. For the meantime I will make a release with those fixed styles and workaround using !important flag in the styles.

If you can provide Plunker example etc. about your issue I could have a look if its actually about the same bug or will it get fixed with those style fixes.

**Comment by sambaptista** [comment]
Finally my issue was a combination of two elements. 

The issue I started in this topic, could be easily avoided by removing the fxFlex attribute and applying a bit of manual written CSS. 

The main issue is not on this aspect (I create another issue to discuss about this). My other usages of `<perfect-scrollbar>` seems to work fine with the actual behavior.

I close as I'm a little bit pressed by the time and would not contribute more to this issue.

**Comment by sconix** [comment]
Ok, then I will just release the tweaks I made. The default styles should then be more reasonable at least.
